### ISSUE TITLE
lsirq/irqtop: please show per-CPU counters

### ISSUE THREAD

**Issue by zeha** [karelzak/util-linux#1109]
Hi,

I was looking into if the new lsirq/irqtop commands can replace irqtop from https://github.com/aabc/ipt-netflow - It appears the big missing piece are per-CPU counters.

For reference, here's a screenshot from ipt-netflow's irqtop:

<img width="1013" alt="Screenshot 2020-07-24 at 00 06 14" src="https://user-images.githubusercontent.com/26175/88343366-88358a80-cd41-11ea-946f-7efdc958b163.png">

Personally I think the per-CPU counters are a lot more interesting, especially with modern hardware using multiple IRQs.

Thanks,
Chris

**Comment by karelzak** [comment]
Well, the initial motivation was to provide a complete picture rather than per-CPU (like we already have in /proc/interrupts) ;-)

Anyway, I can imagine some % utilization columns and --cpus[=<cpuset>] command-line option to display CPU specific data. The question is ho to make it usable for systems with large numbers of CPUs.

**Comment by karelzak** [comment]
Another possibility is to add an extra table with CPUs before the current output (like ipt-netflow in your example). The questions how usable it will be in on systems with many CPUs.
```
irqtop | total: 1192970568 delta: 10059 | ws.net.home | 2021-02-24 13:19:11+01:00

CPU:       0     1
util:    0.1   0.0
%irq:    0.1   0.0

IRQ     TOTAL DELTA NAME
LOC 947987193  7878 Local timer interrupts
TLB  96812016   139 TLB shootdowns
CAL  90565321   694 Function call interrupts
 42  22898064   211 IR-PCI-MSI 1048576-edge nvidia
 30  19765181   987 IR-PCI-MSI 327680-edge xhci_hcd
RES  12446144    92 Rescheduling interrupts
 29   1313329     4 IR-PCI-MSI 512000-edge ahci[0000:00:1f.2]
TRM    575953     0 Thermal event interrupts
 41    215166    49 IR-PCI-MSI 409600-edge eno1
 34     84060     3 IR-PCI-MSI 524289-edge nvme0q1
 40     45212     0 IR-PCI-MSI 524295-edge nvme0q7
 39     44614     1 IR-PCI-MSI 524294-edge nvme0q6
 36     41230     1 IR-PCI-MSI 524291-edge nvme0q3

```

**Comment by karelzak** [comment]
The extra table with CPUs is already implemented in v 2.37.
